### DISCUSSION TITLE
Is filament compatible with mongodb ?

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#1365] [Help] [answered]
Hello, Is filament compatible with mongodb (through jenssegers/laravel-mongodb for example) ?

**Comment by danharrin** [comment, marked as answer]
No.

**Reply by apeisa** [reply]
Filament works very well with the laravel-mongodb, since that library is Eloquent compatible.

I am using it myself and so far only two minor problems where other was solved in recent PR of mine in this repo and other has open PR in laravel-mongodb repo.

If course some features might require SQL, but in those cases mongodb users should just avoid it.

**Reply by apeisa** [reply]
See above @zenati @danharrin

**Reply by nelsondcosta23** [reply]
hey @apeisa how are you?
Im need to deploy a fast backoffice and I have some experience with laravel/filament as my team.
Problem I must use mongoDB. This backoffice its temporary until we have time to setup one in javascript and ts.
This backoffice should be simple. Do you think it worth the risk use the laravel-mongodb ?

**Reply by apeisa** [reply]
It works pretty well. Recently Mongo (company) took the [laravel-mongodb](https://github.com/mongodb/laravel-mongodb) under their account, so it's not just community project anymore.

If your scope is just Filament, then I agree 100% with what @danharrin said. But if you have some other reasons to go for combo of Laravel + Mongo, then I think laravel-mongodb is your best bet and Filament has worked nicely with it.

Anyways, haven't found any other issues with Filament / mongo-db than those that were fixed in 2022, so it has been pretty solid ride for us.

**Reply by danharrin** [reply]
While I am glad it works well for you, while developing Filament I do not consider how a new feature or PR will affect MongoDB users. This could result in breaking changes in patch/minor releases. For example, I might introduce a new SQL statement somewhere to fix an issue.

I have some experience with MongoDB in the past and I really did not enjoy it, so I have basically no interest in spending time supporting it officially in the core. Especially since barely anyone uses it in the Laravel ecosystem, and Laravel themselves don't officially support it with Eloquent.

Hope that makes sense, if you want to be extra safe you should lock your Filament version and upgrade to the next version manually each time it gets released.

**Reply by apeisa** [reply]
I totally understand @danharrin and agree that it is a good decision and statement from your side. Mongo is definitely an outlier on the Laravel ecosystem.
